pkg/app/app.go

```diff
@@ -57,12 +57,12 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 	// TODO(bentheelder): signal handling & timeout
 
 	// ensure the artifacts dir
-	if err := os.MkdirAll(opts.ArtifactsDir(), os.ModePerm); err != nil {
+	if err := os.MkdirAll(opts.RunDir(), os.ModePerm); err != nil {
```

Comment:
we ensure the directory exists, can drop from ginkgo tester.

Reply:
nit: Also update that comment on 59 to match*

Do we ensure it exists with correct permissions?

Reply:
@MushuEE Updated the comment.

@amwat I actually added it to the ginkgo tester because I hit the directory not existing, because the ginkgo tester might be invoked from e.g. [kubetest2-tester-kops](https://github.com/kubernetes/kops/blob/master/tests/e2e/kubetest2-tester-kops/main.go) and doesn't necessarily go through this path (though I'm still learning how this all fits together, so could be wrong here).

Reply:
hmm if you invoke the kubetest2-tester-kops binary directly (not recommended) then you would hit this.
but if you invoke it as `kubetest2 kops --test=kops` then it should go through the kubetest2 lifecycle.

Reply:
/hold cancel

interested in getting this in, can be cleaned up later.